lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_apk_from_aab.rb

```diff
@@ -0,0 +1,132 @@
+module Fastlane
+  module Actions
+    class AndroidGenerateApkFromAabAction < Action
+      def self.generate_command(aab_file_path, apk_output_file_path, keystore_path, keystore_password, keystore_key_alias, signing_key_password)
+        command = "bundletool build-apks --mode universal --bundle #{aab_file_path} --output-format DIRECTORY --output #{apk_output_file_path} "
+        code_sign_arguments = "--ks #{keystore_path} --ks-pass #{keystore_password} --ks-key-alias #{keystore_key_alias} --key-pass #{signing_key_password} "
+        move_and_cleanup_command = "&& mv #{apk_output_file_path}/universal.apk #{apk_output_file_path}_tmp && rm -rf #{apk_output_file_path} && mv #{apk_output_file_path}_tmp #{apk_output_file_path}"
+
+        # Attempt to code sign the APK if a keystore_path parameter is specified
+        command += code_sign_arguments unless keystore_path.nil?
+
+        # Move and rename the universal.apk file to the specified output path and cleanup the directory created by bundletool
+        command += move_and_cleanup_command
+
+        return command
+      end
+
+      def self.run(params)
+        begin
+          sh('command -v bundletool > /dev/null')
+        rescue StandardError
+          UI.user_error!('bundletool is not installed. Please install it using the instructions at https://developer.android.com/studio/command-line/bundletool.')
+          raise
+        end
+
+        # If no AAB param was provided, attempt to get it from the lane context
+        # First GRADLE_ALL_AAB_OUTPUT_PATHS if only one
+        # Second GRADLE_AAB_OUTPUT_PATH if it is set
+        # Else use the specified parameter value
+        if params[:aab_file_path].nil?
+          all_aab_paths = Actions.lane_context[SharedValues::GRADLE_ALL_AAB_OUTPUT_PATHS] || []
+          aab_file_path = if all_aab_paths.count == 1
+                            all_aab_paths.first
+                          else
+                            Actions.lane_context[SharedValues::GRADLE_AAB_OUTPUT_PATH]
+                          end
+        else
+          aab_file_path = params[:aab_file_path]
+        end
+
+        # If no AAB file path was found, raise an error
+        if aab_file_path.nil?
+          UI.user_error!('No AAB file path was specified and none was found in the lane context. Please specify the `aab_file_path` parameter or ensure that the relevant build action has been run prior to this action.')
+          raise
+        end
+
+        apk_output_file_path = params[:apk_output_file_path]
+        keystore_path = params[:keystore_path]
+        keystore_password = params[:keystore_password]
+        keystore_key_alias = params[:keystore_key_alias]
+        signing_key_password = params[:signing_key_password]
+
+        sh(generate_command(aab_file_path, apk_output_file_path, keystore_path, keystore_password, keystore_key_alias, signing_key_password))
+      end
```

Comment:
If we go with suggestion 2️⃣ , we might want to return the path the the generated `apk` (given that if `apk_output_file_path` wasn't provided at call site we'll infer it, but that means that if the call site needs that APK output path it's nice to be able to get it as return value)

```suggestion
        apk_output_file_path
      end
```

If we do that, we should also document it by implementing:
```ruby
def self.return_type
  :string
end

def self.return_value
  'The path to the APK that has been generated'
end
```

Reply:
Implemented via [this line](https://github.com/wordpress-mobile/release-toolkit/pull/467/files#diff-89bdefd7112342eae6bc39216c324717a41f958babd4374b1bc30e3a47a7c0b1R35) to return the value + [this action documentation](https://github.com/wordpress-mobile/release-toolkit/pull/467/files#diff-89bdefd7112342eae6bc39216c324717a41f958babd4374b1bc30e3a47a7c0b1R132-R138).